surveys/promise-rejections/survey.md

```diff
@@ -0,0 +1,235 @@
+  - [ ] `try / catch` wrapping an `await` operation
+  - [ ] Leave the handling to my caller
+  - [ ] Not writing Promise-based code or using Promise-based libraries
+
+## Do you know that Node.js has a global handler for unhandled rejections (`process.on('unhandledRejection')`)? If so, do you use it?
+ 
+  - [ ] I use `process.on('unhandledRejection')`
+  - [ ] I don't use `process.on('unhandledRejection')`
+  - [ ] I didn't know `process.on('unhandledRejection')` existed
+
+## Do you know that Node.js has a flag to change the behavior of unhandled rejections `--unhandled-rejections`? If so, do you use it?
+
+  - [ ] I use `--unhandled-rejections` set to `'strict'`
+  - [ ] I use `--unhandled-rejections` set to `'warn'`
+  - [ ] I use `--unhandled-rejections` set to `'none'`
+  - [ ] I don't use `--unhandled-rejections`
+  - [ ] I use a third-party library (like make-promises-safe) to deal with unhandled rejections
+  - [ ] I didn't know `--unhandled-rejections` existed
+
+
+## Do you know that Node.js has a global handler for uncaught exception (`process.on('uncaughtException')`)? If so, do you use it?
+ 
+  - [ ] I use `process.on('unhandledRejection')`
+  - [ ] I don't use `process.on('unhandledRejection')`
+  - [ ] I didn't know `process.on('unhandledRejection')` existed
+
+Are you using Promises in any of the following kinds of applications? (check all that apply)
+
+  - [ ] Production-level code
+  - [ ] Tests
+  - [ ] Benchmarks
+  - [ ] Build pipeline / infrastructure
+  - [ ] Examples and Demos
+  - [ ] Other (please describe)
+
+## Which of the use cases described below have you spent most of your time on?
+
+  - [ ] Third-party libraries
+  - [ ] Command-line tools
+  - [ ] Web servers
+  - [ ] Computation-heavy processing
+  - [ ] Desktop applications
+  - [ ] Other (please elaborate)
+
+## When a rejected promise doesn't have a catch handler, what is the behavior of your application today? 
+
+For example:
+
+```js
+async function error() {
+  throw new Error();
+}
+error();
+```
+
+If working on multiple applications or projects, choose the answer that describes the behavior on your biggest project, or on the project you worked the most.
+
+  - [ ] Node.js logs a warning alongside a deprecation notice, execution continues (this is the default Node.js behavior today)
+  - [ ] Logs a warning, continue running, no deprecation warning
+  - [ ] Logs a warning, continue running, no deprecation warning, exits with an error code when program finishes
+  - [ ] Exit as soon as possible (this is the default Node.js behavior for uncaught exceptions today)
+  - [ ] Other (please elaborate)
+
+## How do you deal with managing resources wrapped with promises when an unhandled rejection occurs?
+
+Node.js code may wrap resource managing code in async/await:
+
+```js
+// if this method wasn't async, node would crash by default
+myEmitter.on('event', async () => {
+   // react to event
+  databaseConnection.release(); // throws an error
```

Comment:
I don't understand the comment about throwing an error. If the method to "release" the database connection throws then either A) It released anyway or B) its impossible to release the connection

Maybe you meant a different example where an error is thrown before `release()` and the `release()` method is never called.

```js
await databaseConnection.getValue() // throws an error
databaseConnection.release()
```

Reply:
@benjamingr suggested this example, so I'll let him answer.

Reply:
> A) It released anyway or B) its impossible to release the connection

Either releasing the connection failed (in which way you probably want to restart the server) or there exists a more "severe" release API for example:

```
database.release(); // EREAD_IN_PROGRESS
database.forceRelease(); // works
```

I think it's also fine for the example to show release never being called (by throwing an error earlier) and it might simplify things.

Reply:
Adding `// throws an error; oops forgot to call database.forceRelease()`

Would resolve my confusion.

Reply:
Let's go with the getValue example above ("release never called"), I think it simplifies and clarifies things.